src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -618,10 +592,42 @@ public void onClick(DialogInterface dialog, int which) {
                 }
                 return true;
             }
+            case R.id.action_sort_by_date:
+                item.setChecked(true);
+                sortByDate(PreferenceManager.getSortAscending(this));
+                return true;
+            case R.id.action_sort_by_name:
+                item.setChecked(true);
+                sortByName(PreferenceManager.getSortAscending(this));
+                return true;
+            case R.id.action_sort_by_size:
+                item.setChecked(true);
+                sortBySize(PreferenceManager.getSortAscending(this));
+                return true;
             default:
-                retval = super.onOptionsItemSelected(item);
         }
-        return retval;
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void recoverSortMenuFormPreferences(Menu menu) {
+        // setup sort menu
+        if(menu != null && mDescendingMenuItem != null) {
+            mDescendingMenuItem = menu.findItem(R.id.action_sort_descending);
```

Comment:
I think you don't need this initialization since it was done [here](https://github.com/owncloud/android/pull/2226/files#diff-845104e37d2ed427df8ae96910082d31R532)